runtime/android/core_internal/src/org/xwalk/core/internal/XWalkViewDelegate.java

```diff
@@ -145,13 +153,73 @@ private static void init(final Context context) {
             CommandLine.init(readCommandLine(context.getApplicationContext()));
         }
 
-        ResourceExtractor.setMandatoryPaksToExtract(MANDATORY_PAKS);
-        int resListResId = context.getResources().getIdentifier(
+        final boolean isSharedMode = !context.getPackageName().equals(
+                context.getApplicationContext().getPackageName());
+
+        HashMap<String, ResourceEntry> resourceList = new HashMap<String, ResourceEntry>();
+        if (isSharedMode) {
+            for (String resource : MANDATORY_PAKS) {
+                resourceList.put(resource, new ResourceEntry(0, "", resource));
+            }
+        } else {
+            int resListId = context.getResources().getIdentifier(
                 XWALK_RESOURCES_LIST_RES_NAME, "array", context.getClass().getPackage().getName());
-        if (resListResId == 0) {
-            resListResId = context.getResources().getIdentifier(
+            if (resListId == 0) {
+                resListId = context.getResources().getIdentifier(
                     XWALK_RESOURCES_LIST_RES_NAME, "array", context.getPackageName());
+            }
+
+            try {
```

Comment:
Yes, the old code had this check. But if the check faild, it did nothing except logging the error. I don't think that was effective error-handling, because it won't help if the error really happened. In new code, I call `Assert.fail()` for this kind of situations. Because missing the resource file should never happen. And if it really happen, we can notice that right the first time of our develpment.